src/aws-elasticache/user.ts

```diff
@@ -0,0 +1,334 @@
+   */
+  readonly userId: string;
+}
+
+/**
+ * A new or imported user group.
+ */
+export abstract class UserBase extends Resource implements IUser {
+  /**
+   * Imports an existing User from attributes
+   */
+  public static fromUserId(scope: Construct, id: string, userId: string): IUser {
+    class Import extends UserBase implements IUser {
+      public readonly userId = userId;
+      public readonly userArn = Stack.of(this).formatArn({
+        resource: 'elasticache',
+        service: 'user',
+        resourceName: userId,
+      });
+    }
+    return new Import(scope, id);
+  }
+
+  /**
+   * The ARN of the user.
+   */
+  public abstract readonly userArn: string;
+
+  /**
+   * The ID of the user.
+   */
+  public abstract readonly userId: string;
+
+  /**
+   * Grant the given identity the specified actions
+   * @param grantee the identity to be granted the actions
+   * @param actions the data-access actions
+   *
+   * @see https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonelasticache.html
+   */
+  public grant(grantee: aws_iam.IGrantable, ...actions: string[]): aws_iam.Grant {
+    return aws_iam.Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: [
+        Stack.of(this).formatArn({
+          service: 'elasticache',
+          resource: 'user',
+          resourceName: this.userId,
+          arnFormat: ArnFormat.COLON_RESOURCE_NAME,
+        }),
+      ],
+    });
+  }
+
+  /**
+   * Permits an IAM principal to perform connect to the user.
+   *
+   * Actions: Connect
+   *
+   * @param grantee The principal to grant access to.
+   * @see https://docs.aws.amazon.com/AmazonElastiCache/latest/dg/IAM.IdentityBasedPolicies.html#iam-connect-policy
+   */
+  public grantConnect(grantee: aws_iam.IGrantable): aws_iam.Grant {
+    return this.grant(grantee, 'elasticache:Connect');
+  }
+}
+
+/**
+ * Represents an User construct in AWS CDK.
+ *
+ * @example
+ *
+ * const user = new User(
+ *   stack,
+ *   'User',
+ *   {
+ *     authenticationType: AuthenticationType.IAM,
+ *   },
+ * );
+ */
+export class User extends UserBase implements IUser {
+  /**
+   * The ARN of the user.
+   */
+  readonly userArn: string;
+
+  /**
+   * The ID of the user.
+   */
+  readonly userId: string;
+
+  private readonly props: UserProps;
+
+  constructor(scope: Construct, id: string, props: UserProps) {
+    super(scope, id, {
+      physicalName:
+        props.userId ??
+        Lazy.string({
+          produce: () => Names.uniqueResourceName(this, { maxLength: 40 }).toLowerCase(),
+        }),
+    });
+    this.props = props;
+
+    this.validateUserId();
+    this.validateUserName();
+    this.validateAuthenticationSettings();
+
+    const user = this.createResource(this, 'Resource', {
+      engine: Engine.REDIS,
+      userId: this.physicalName,
+      userName: this.props.userName ?? this.physicalName,
+      accessString: this.props.accessString ?? 'off -@all',
```

Comment:
This property is optional in CloudFormation, but if omitted, it results in an internal error during deployment. 
Therefore, the default value that is automatically set when creating from the management console is used.

`off -@all` disables all commands for the specified user. 

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticache-user.html#cfn-elasticache-user-accessstring